build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs

```diff
@@ -21,7 +21,8 @@ protected override async Task<bool> Execute (Context context)
 			var dotnetPath = Configurables.Paths.DotNetPreviewPath;
 			dotnetPath = dotnetPath.TrimEnd (new char [] { Path.DirectorySeparatorChar });
 
-			if (!await InstallDotNetAsync (context, dotnetPath, BuildToolVersion)) {
+			if (!await InstallDotNetAsync (context, dotnetPath, BuildToolVersion, useCachedInstallScript: true) &&
```

Comment:
Is there a better way of doing this? Can we not get the hash or crc32 of the file we want to download and compare it to the cached version? 

using ` curl -I https://raw.githubusercontent.com/dotnet/install-scripts/refs/heads/main/src/dotnet-install.sh`

We get an `etag` field we might be able to use to compare.

Reply:
eg etag: "b2ba8c3f17e539ba16047ec26d7eb3cb58841bdc40aa0a27aab055feab567d1b"

Reply:
We could store the etag locally and compare it

Reply:
oh it does `etag: 0x8DD34C45DC46329` but its different